cmd/kar-controllers/app/server.go

```diff
@@ -76,14 +84,69 @@ func Run(opt *options.ServerOption) error {
 }
 
 // Starts the health probe listener
-func listenHealthProbe(opt *options.ServerOption) error {
-	handler := http.NewServeMux()
-	handler.Handle("/healthz", &health.Handler{})
-	err := http.ListenAndServe(opt.HealthProbeListenAddr, handler)
-	if err != nil {
-		return err
-	}
-
-	return nil
+func startHealthAndMetricsServers(opt *options.ServerOption) error {
+    // Start the metrics server
+    RecordMetrics()
+
+    metricsHandler := http.NewServeMux()
+    metricsHandler.Handle("/metrics", promhttp.Handler())
```

Comment:
It would be good to add the default Prometheus error handler (`ErrorHandling: promhttp.HTTPErrorOnError`)